src/nexusrpc/handler/_common.py

```diff
@@ -1,13 +1,43 @@
 from __future__ import annotations
 
-from abc import ABC
+from abc import ABC, abstractmethod
 from collections.abc import Mapping, Sequence
 from dataclasses import dataclass, field
 from typing import Any, Generic, Optional
 
 from nexusrpc._common import Link, OutputT
 
 
+class OperationTaskCancellation(ABC):
+    """
+    Indicates whether a a Nexus task has been cancelled during a sync operation or before an async operation has
+    returned a token.
+
+    Nexus worker implementations are expected to provide an implementation that enables
+    cooperative cancellation for both sync and async operation handlers.
+    """
```

Comment:
Not important, but may want to add doc explaining these values are set before asyncio.cancel is performed and `wait_until_X` methods are therefore racy with it.